File: app/src/main/java/com/example/mrakopediareader/pageslist/HOTMList.java
package com.example.mrakopediareader.pageslist;

import com.example.mrakopediareader.api.API;
import com.example.mrakopediareader.api.dto.Page;

import java.util.ArrayList;
import java.util.Optional;

import io.reactivex.rxjava3.core.Observable;

public class HOTMList extends PagesList {
    @Override
    protected Observable<ArrayList<Page>> getPages() {
        return Optional
                .ofNullable(api)
                .map(API::getHOTM)
                .orElse(Observable.just(new ArrayList<>()));
    }
}

File: app/src/main/java/com/example/mrakopediareader/pageslist/HOTMList.kt
package com.example.mrakopediareader.pageslist

import com.example.mrakopediareader.api.dto.Page
import io.reactivex.rxjava3.core.Observable

class HOTMList : PagesList() {
    override fun getPages(): Observable<List<Page>> {
        return api.getHOTM()
    }
}